File: src/#94/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	ans := []int{}
	traversal(root, &ans)
	return ans
}

func traversal(root *TreeNode, arr *[]int) {
	if root == nil {
		return
	}

	traversal(root.Left, arr)
	*arr = append(*arr, root.Val)
	traversal(root.Right, arr)
}

File: src/#94/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var inorderTraversal = function (root) {
  let arr = []

  function traversal(root = null) {
    if (root === null) {
      return
    }

    traversal(root.left)
    arr.push(root.val)
    traversal(root.right)
  }
  traversal(root)

  return arr
}
